### ISSUE TITLE
Support an SDK feature list to become source of truth

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-release-registry#39]
The goal is for SDKs to report their feature set via the release registry. Adding features to the registry can become part of the developer workflow (i.e add a features.yaml to the repo root and craft appends to the registry). 

The goal is to remove sheets with manually mapped feature matrix. And to have a generated page listing SDKs and the features they support.

Follow up with @PeloWriter about consuming this in the docs.

**Comment by BYK** [comment]
So getsentry/sentry#23585 essentially needs this feature to be completed.

@bruno-garcia can I assign that PR to you or is there another point person for this work?

**Comment by bruno-garcia** [comment]
We are not able to schedule this right now so it needs to stay unassigned.
Unless you want to pick it up? :)

**Comment by BYK** [comment]
>We are not able to schedule this right now so it needs to stay unassigned.
>Unless you want to pick it up? :)

No, thank you my plate is quite full and I don't think I have full context around this yet.

I'd rather add `Status: On Hold` to indicate we want to work on this but not yet to prevent it from being closed by stale bot.

**Comment by PeloWriter** [comment]
This would be very useful for our customers.

**Comment by PeloWriter** [comment]
The goal here is to programmatically answer the question "does this SDK support this feature?" which would ensure the content is correct and current. We would then also link to the associated content. 

Here's one example (from React Native):
Features:

- Automatic Native Crash [Error Tracking](/product/error-monitoring/) (using both `sentry-cocoa` & `sentry-android`)
- Offline storage of events
- Events [enriched](/platforms/react-native/enriching-events/context/) with device data
- [Autolinking](https://reactnative.dev/blog/2019/07/03/version-60#native-modules-are-now-autolinked)
- [Breadcrumbs](/platforms/react-native/enriching-events/breadcrumbs/) created for outgoing `http` request with XHR and Fetch; UI and system events; and console logs
- [Release Health](/product/releases/health/) tracks crash free users and sessions
- [Performance Monitoring](/product/performance/) creates transactions automatically for React Navigation v4 and v5; creates spans automatically for XHR and Fetch
- Under the hood the SDK relies on our [JavaScript SDK](/platforms/javascript/), which makes all functions available for JavaScript also available in this SDK
- On Device symbolication for JavaScript (in Debug)
- [RAM bundle support](/platforms/react-native/manual-setup/ram-bundles/)
- [Hermes support](/platforms/react-native/manual-setup/hermes/)
- [Expo support](https://docs.expo.io/guides/using-sentry/)

**Comment by PeloWriter** [comment]
@bruno-garcia - can we get a definitive list of all possible features to get this started?

**Comment by bruno-garcia** [comment]
I wonder if we can have a "link to examples" as a feature?
For example, in the `sentry-react-native` repo we'd have:

```
  - example-url: "https://github.com/getsentry/examples/tree/master/react-native"
```

**Comment by bruno-garcia** [comment]
Related prior work: https://github.com/getsentry/sentry-java/pull/1527
